core/src/main/java/com/amplifyframework/auth/AuthCategoryBehavior.java

```diff
@@ -86,7 +86,7 @@ void resendSignUpCode(
     void signIn(
             @Nullable String username,
             @Nullable String password,
-            @NonNull AuthSignInOptions options,
+            @Nullable AuthSignInOptions options,
```

Comment:
Instead of doing this, It would be better to provide an overloaded method that does not require any `options`.

If your options as a user are to call:
```
void signIn(@Nullable first, @Nullable last, @Nullable options)
```
as:
```
signIn(null, null, null, ok -> {}, notOk -> {})
```
Wouldn't you rather have just done:
```
signIn(ok -> {}, notOk -> {})
```

The null params end up snaking down through the code base and creating a lot of branch complexity once everything has a `if (thing == null)`  around it.

So, the way I would handle this is to have two copies of sign-in, one that requires some non-null params, and one that does not take in those params, at all.

Internal to the implementation of the second method, choose "sane defaults." `AuthSignInOptions.defaults()` or something, that you can proxy into the other method, which _does_ need the anon-null param.

Reply:
So I actually do have an overloaded method that does not require any options. I made this Nullable so the method that didn't take options could call this with null options since the options aren't the kind which take default values - they either exist or they don't. But if you feel strongly about this, I can have it pass a blank options object over to the other method instead and change this back to NonNull.